### DISCUSSION TITLE
I want to get both count and rows, can I clone a query?

### DISCUSSION THREAD

**Discussion by lz1998** [SeaQL/sea-orm#368] [Q&A] [answered]
I want to get both count and rows, can I clone a query?

Is there any better solution?
```rust
    let conn = get_conn().await;
    let mut q = Entity::find();
    
    let q = q.filter(...);
    let count = q.clone().count().await;
    let rows = q.offset(...).limit(...).all().await;
```

**Comment by billy1624** [comment, marked as answer]
Hey @lz1998, you could use paginator to do both without cloning the query.
- Set the page size with `.paginate()` and offset it with `.fetch_page()`
- Count number of rows by calling `.num_items()`

https://docs.rs/sea-orm/0.4.1/sea_orm/struct.Paginator.html
